py/torch_tensorrt/dynamo/conversion/impl/normalization/ops.py

```diff
@@ -36,59 +36,136 @@ def batch_norm(
     input: TRTTensor,
     weight: Optional[Union[torch.Tensor, np.ndarray]],
     bias: Optional[Union[torch.Tensor, np.ndarray]],
-    running_mean: Optional[Union[torch.Tensor, np.ndarray]],
-    running_var: Optional[Union[torch.Tensor, np.ndarray]],
+    running_mean: Union[TRTTensor, Optional[Union[torch.Tensor, np.ndarray]]],
+    running_var: Union[TRTTensor, Optional[Union[torch.Tensor, np.ndarray]]],
     training: bool,
     momentum: float,
     eps: float,
     cudnn_enabled: bool,
     return_mean_rstd: bool,
 ) -> Union[TRTTensor, Tuple[TRTTensor, torch.Tensor, torch.Tensor]]:
+
     if has_dynamic_shape(input.shape):
         assert input.shape[1] != -1, "Channel dim can't be dynamic for batch norm."
 
-    if weight is None:
-        weight = 1.0
+    # Save the original output shape for later use
+    output_shape = input.shape
 
-    if bias is None:
-        bias = 0.0
+    # Handle case when running_mean or running_var is TRTTensor
+    if isinstance(running_mean, TRTTensor) or isinstance(running_var, TRTTensor):
+        # Default values if weight, bias, running_mean, running_var are None
+        if weight is None:
+            weight = get_trt_tensor(ctx, 1.0, f"{name}_weight", input.dtype)
+        if bias is None:
+            bias = get_trt_tensor(ctx, 0.0, f"{name}_bias", input.dtype)
+        if running_mean is None:
+            running_mean = get_trt_tensor(ctx, 0.0, f"{name}_running_mean", input.dtype)
+        if running_var is None:
+            running_var = get_trt_tensor(ctx, 1.0, f"{name}_running_var", input.dtype)
```

Comment:
1) Do we need to cast these parameters to the type of `input`? I think there's probably a case that `input` is int type while `weight`, `bias`, `running_mean`, and/or `running_var` are float type. It seems problematic to force cast float to int. The `dtype` is optional so you can just leave it blank.

2) `weight` and `bias` could be ITensor as well right?

Reply:
>>Do we need to cast these parameters to the type of input
I thinks this is required for [strongly-typed-networks](https://docs.nvidia.com/deeplearning/tensorrt/developer-guide/index.html#strongly-typed-networks), different type can be allowed for weak typed network.
I noticed it by enabling strongly typed networks with some model. Here is [some changes](https://github.com/keehyuna/TensorRT/commit/2e32bde7370bf26d8f7b7112b8aa53b52ab39644#diff-47e9eb66cba97ff59bf6c1355e0506921be5780f8796d725f9dc3abcdb22cda5R127) to keep same type in ops. I only saw float32 and half float, I'm now sure if float and int tensor in same layer is possible.

Reply:
->>Do we need to cast these parameters to the type of input?
If `input_val` is a `TRTTensor`, the input is returned unchanged ([code](https://github.com/pytorch/TensorRT/blob/db15ecaa6f18cfdae99e14b5ff497a60abcf231b/py/torch_tensorrt/dynamo/conversion/converter_utils.py#L386)), so the type stays the same. If `input_val` is something else, `create_constant` converts the type using `to_numpy` ([code](https://github.com/pytorch/TensorRT/blob/db15ecaa6f18cfdae99e14b5ff497a60abcf231b/py/torch_tensorrt/dynamo/conversion/converter_utils.py#L344)). However, if the `value` is an `np.ndarray` or `torch.Tensor`, it keeps the original type ([code](https://github.com/pytorch/TensorRT/blob/db15ecaa6f18cfdae99e14b5ff497a60abcf231b/py/torch_tensorrt/dynamo/conversion/converter_utils.py#L550) and [code](https://github.com/pytorch/TensorRT/blob/db15ecaa6f18cfdae99e14b5ff497a60abcf231b/py/torch_tensorrt/dynamo/conversion/converter_utils.py#L559)). This means setting `input.dtype` has no effect.

I confirmed this behavior when I removed the `input.dtype` argument.

As for the issue @keehyuna  mentioned, it's new to me, so I'll investigate.

->> `weight` and `bias` could be ITensor as well right?
Additionally, according to the schema, `weight` and `bias` can't be `ITensor`, but the converter works fine. I've tested it, and it works successfully.

Reply:
> ->> weight and bias could be ITensor as well right?
Additionally, according to the schema, weight and bias can't be ITensor, but the converter works fine. I've tested it, and it works successfully.

@chohk88 Did I miss something? The schema is: 
```
- func: batch_norm(Tensor input, Tensor? weight, Tensor? bias, Tensor? running_mean, Tensor? running_var, bool training, float momentum, float eps, bool cudnn_enabled) -> Tensor
- func: _native_batch_norm_legit_no_training(Tensor input, Tensor? weight, Tensor? bias, Tensor running_mean, Tensor running_var, float momentum, float eps) -> (Tensor, Tensor, Tensor)
```

My understanding to `Tensor? weight, Tensor? bias` is that they could be None, ITensor, torch.Tensor, or np.ndarray. Please correct me if I'm wrong.

Reply:
Oh, I misunderstood that. You're right—`weight`, `bias`, `running_mean`, and `running_var` can all be `TRTTensor`. I’ve combined the separate converter and added some test cases for this.